chain/chain/src/runtime/metrics.rs

```diff
@@ -36,6 +36,44 @@ pub(crate) static PREPARE_TX_SIZE: Lazy<HistogramVec> = Lazy::new(|| {
     .unwrap()
 });
 
+pub(crate) static PREPARE_TX_GAS: Lazy<HistogramVec> = Lazy::new(|| {
+    try_create_histogram_vec(
+        "near_prepare_tx_gas",
+        "How much gas was spent for processing new transaction.",
+        &["shard_id"],
+        // 100e9 = 100 Ggas
+        // A transaction with no actions costs 108 Ggas to process.
+        // A typical function call costs ~300 Ggas.
+        // The absolute maximum is defined by `max_tx_gas` = 500 Tgas.
+        // This ranges from 100 Ggas to 409.6 Tgas as the last bucket boundary.
+        Some(exponential_buckets(100e9, 2.0, 12).unwrap()),
+    )
+    .unwrap()
+});
+
+pub(crate) static CONGESTION_PREPARE_TX_REJECTED: Lazy<HistogramVec> = Lazy::new(|| {
+    try_create_histogram_vec(
+        "near_congestion_prepare_tx_rejected",
+        "The number of transactions removed from the transaction pool due to congestion.",
+        &["shard_id"],
+        // Histogram boundaries are inclusive. Pick the first boundary below 1
+        // to have 0 values as a separate bucket.
+        // In exclusive boundaries, this would be equivalent to:
+        // [0, 10, 100, 1_000, 10_000]
+        Some(exponential_buckets(0.99999, 10.0, 6).unwrap()),
+    )
+    .unwrap()
+});
```

Comment:
Totally optional - this could be generalized to PREPARE_TX_REJECTED by adding an extra label "reason" where reasons can be `congestion`, `chain_validate`, `verify_and_charge_transaction` (but better names).

Reply:
Totally optional - PREPARE_TX_COUNT may also be interesting.

Reply:
Very nice idea with the reason label. I've added it and tried to come up with meaningful names. Please take a look if the names make sense.

For `PREPARE_TX_COUNT`, I see that it could be interesting but I am unsure about adding it here. My first instinct is that this should be a metric not on prepare_tx, but rather on when we apply a chunk. All of the tx we include will always be executed when the chunk is applied, so the number of executed transactions would be the same as the one prepared.

So, I checked what metrics we have on transaction execution. I only found `TRANSACTION_PROCESSED_TOTAL`.

```rust
pub static TRANSACTION_PROCESSED_TOTAL: Lazy<IntCounter> = Lazy::new(|| {
    try_create_int_counter(
        "near_transaction_processed_total",
        "The number of transactions processed since starting this node",
    )
    .unwrap()
});
```

This does not count per shard, so it's not really a replacement for your suggestion. Or do you think looking at the difference in `TRANSACTION_PROCESSED_TOTAL` would give you the visibility you hope to get from `PREPARE_TX_COUNT`?

Of course, this information is also included in the chunk itself, just not exposed as metric.

Reply:
> Please take a look if the names make sense.

LGTM

> TRANSACTION_PROCESSED_TOTAL

TRANSACTION_PROCESSED_TOTAL seems like it's counting transactions across blocks and shards so it's different than my original suggestion. No need to add anything now, if we find it lacking in tests we can always come back and it.